### ISSUE TITLE
`scripted` does not publishLocal transitive project dependencies

### ISSUE THREAD

**Issue by jastice** [sbt/sbt#3248]
## steps

For example:
```
val lib = project
val plugin = project.dependsOn(lib) // an sbt plugin
```

1. Define a scripted test for `plugin`
2. Run `scripted` from sbt

## problem

1. scripted does a `publishLocal` only on `plugin`, dependency is not published
2. test run fails because of missing dependency
3. workaround: I can manually do publishLocal := publishLocal.dependsOn(publishLocal in lib).value

sbt version: 0.13.15

**Comment by olafurpg** [comment]
Have you tried?

```
val lib = project
val plugin = project.dependsOn(lib).aggregate(lib)
```

That seems to publish lib + plugin on `plugin/publishLocal`

**Comment by eed3si9n** [comment]
In terms of the expectation of `scripted` and `publishLocal`, I would suggest:

- `scripted` works out of the box by calling publishLocal recursively along the subproject dependencies.
- `publishLocal` works, but it's limited to the current subproject, unless `.aggregate` is specified.

**Comment by jastice** [comment]
This seems reasonable

**Comment by dwijnand** [comment]
Tweaked the subject to reflect this is `scripted`-related.

**Comment by solarmosaic-kflorence** [comment]
Just ran into this on SBT `1.4.4` and thought it was not very intuitive. I would expect the `.dependsOn` to get the appropriate dependencies published with `publishLocal`.

**Comment by solarmosaic-kflorence** [comment]
FWIW I just updated my project to `1.5.2` and was able to remove the `publishLocal := publishLocal.dependsOn(publishLocal in lib).value`

**Comment by eed3si9n** [comment]
Thanks Kyle. This was in fact fixed in 1.5.0 by @steinybot - https://github.com/sbt/sbt/pull/6351
